### PR TITLE
[FIX] sale_stock: procurement qty based on location

### PR DESCRIPTION
**Pull request by amoyaux**
Close #28315

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by pimodoo** [comment]
Ca c'est beau! ('This is Beautiful!')

**Comment by pimodoo** [comment]
@sle-odoo Check it and call the robot

**Comment by wtaferner** [comment]
@amoyaux @sle-odoo 
Could we please finally merge this?

**Comment by wtaferner** [comment]
@sle-odoo @nim-odoo 
Gentle reminder that this is still not merged in stable...

OPW-1911019 was cancelled without answer?!
I wonder that nobody is experiencing issues, but yeah, some corners are obviously not recognized. Most likely because the battlefield is huge :wink:

**Comment by wtaferner** [comment]
@amoyaux @pimodoo @jco-odoo 
This is still not merged? I wonder that no one was hit hard by this pitfall yet?  

Look at #28315 which was closed in favor of this...please @sle-odoo check it like @pimodoo asked for and relay to the little merge god to fulfill its job :wink:

File: addons/sale_stock/models/sale_order.py [hunk 1]
```diff
@@ -224,13 +224,24 @@ def _prepare_procurement_values(self, group_id=False):
...
+        for move in incoming_moves:
+            qty -= move.product_uom._compute_quantity(move.product_uom_qty, self.product_uom)
         return qty
 
+    def _get_outgoing_incoming_moves(self):
+        outgoing_moves = self.env['stock.move']
+        incoming_moves = self.env['stock.move']
+        for move in self.move_ids.filtered(lambda r: r.state != 'cancel' and not r.scrapped):
+            if move.location_dest_id.usage == "customer":
+                if not move.origin_returned_move_id or (move.origin_returned_move_id and move.to_refund):
```
Comment (jco-odoo) [inline addons/sale_stock/models/sale_order.py]: This line does not make sense to me.  Should not it be similar to the stock_account incoming/outgoing?

**Comment by jco-odoo** [comment]
@sle-odoo @amoyaux It is not good to let it depend on picking type code.  (meant just for reports/views)

**Comment by sle-odoo** [comment]
i would retarget this branch to master, nobody complained
in master i think there's just a patch in _get_qty_procurement to do to follow the convention of  _compute_qty_delivered

**Comment by wtaferner** [comment]
@sle-odoo 
Haha, thank you for calling me nobody, but honestly this is nothing to fix in master, it does actually fuck up data and I do not like the fact that I need to patch this for v12 too.

**Comment by sle-odoo** [comment]
adapted on master + a test in https://github.com/odoo/odoo/pull/35257

**Pull request closed without merging**